cms/djangoapps/contentstore/views/course.py

```diff
@@ -1460,32 +1460,108 @@ def _get_usage_info(store, course, split_tests):
+                    'url': 'url_to_unit_1'
+                },
+                {
+                    'label': 'Unit 2 / Problem 2',
+                    'url': 'url_to_unit_2'
+                }
+            ],
+        }
+        """
+        usage_info = {}
+        for module in modules:
+            if hasattr(module, 'group_access') and module.group_access:
+                (__, group_ids), = module.group_access.items()
+                for group_id in group_ids:
+                    if group_id not in usage_info:
+                        usage_info[group_id] = []
+
+                    module_parent = module.get_parent()
+
+                    if not module_parent:
+                        log.warning("Parent module of content_groups not found: %s", module.location)
+                        continue
+
+                    unit_url = reverse_usage_url(
+                        'container_handler',
+                        course.location.course_key.make_usage_key(
+                            module_parent.location.block_type,
+                            module_parent.location.name
+                        )
+                    )
+
+                    usage_info[group_id].append({
+                        'label': u"{} / {}".format(module_parent.display_name, module.display_name),
+                        'url': unit_url,
+                    })
+        return usage_info
+
+    @staticmethod
     def update_usage_info(store, course, configuration):
         """
         Update usage information for particular Group Configuration.
 
         Returns json of particular group configuration updated with usage information.
         """
-        # Get all Experiments that use particular Group Configuration in course.
-        split_tests = store.get_items(
-            course.id,
-            category='split_test',
-            content={'user_partition_id': configuration.id}
-        )
-        configuration_json = configuration.to_json()
-        usage_information = GroupConfiguration._get_usage_info(store, course, split_tests)
-        configuration_json['usage'] = usage_information.get(configuration.id, [])
+        configuration_json = None
+        # Get all Experiments that use particular  Group Configuration in course.
+        if configuration.scheme.name == 'random':
+            split_tests = store.get_items(
+                course.id,
+                category='split_test',
+                content={'user_partition_id': configuration.id}
+            )
+            configuration_json = configuration.to_json()
+            usage_information = GroupConfiguration._get_usage_info(store, course, split_tests)
+            configuration_json['usage'] = usage_information.get(configuration.id, [])
+        elif configuration.scheme.name == 'cohort':
+            # In case if scheme is "cohort"
+            configuration_json = GroupConfiguration.update_content_group_usage_info(store, course, configuration)
```

Comment:
I don't think so that we can do this in shared code because the usage information structure is different for both. For **cohort** we need to embed the usage info. into particular group and in case of **experiment** we have to inject the info. for whole configuration.